### ISSUE TITLE
Error when visiting face thumbnail URL from different user

### ISSUE THREAD

**Issue by Derkades** [matiasdelellis/facerecognition#352]
This is a bit of a weird one, but it should probably be fixed.

### Expected behaviour

Permission denied error

### Actual behaviour

Internal server error (see log below)

### Steps to reproduce
1. Visit yourcloud/settings/user/facerecognition with network logging open in browser
2. Copy a thumbnail URL
4. Visit thumbnail URL and confirm it works (shows a picture)
5. Log in as different nextcloud user
6. Visit thumbnail URL again

### Server configuration

* **Operating system:** debian

* **Pdlib version:** 1.0.2

* **How is DLib installed:** From source, working with test suite

* **How is PDlib installed:** From source, working with test suite

* **PHP version:** 7.4.11

* **Web server:** Nginx

* **Database:** MySQL

* **Nextcloud version:** 19.0.4

### Client configuration

* **Browser:** Firefox 82

* **Operating system:** Ubuntu 20.10

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
[index] Error: OCP\AppFramework\Db\DoesNotExistException: Did expect one result but found none when executing: query "SELECT `id`, `file`, `is_processed`, `error`, `last_processed_time`, `processing_duration` FROM `*PREFIX*facerecog_images` `i` WHERE (`user` = :dcValue1) AND (`id` = :dcValue2)";  at <<closure>>

 0. /var/www/html/lib/public/AppFramework/Db/QBMapper.php line 334
    OCP\AppFramework\Db\QBMapper->findOneQuery(OC\DB\QueryBuilder\QueryBuilder {})
 1. /var/www/html/apps/facerecognition/lib/Db/ImageMapper.php line 52
    OCP\AppFramework\Db\QBMapper->findEntity(OC\DB\QueryBuilder\QueryBuilder {})
 2. /var/www/html/apps/facerecognition/lib/Controller/FaceController.php line 88
    OCA\FaceRecognition\Db\ImageMapper->find("Test", 11799)
 3. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 170
    OCA\FaceRecognition\Controller\FaceController->getThumb("2621", "128")
 4. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 100
    OC\AppFramework\Http\Dispatcher->executeController(OCA\FaceRecognit ... {}, "getThumb")
 5. /var/www/html/lib/private/AppFramework/App.php line 137
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\FaceRecognit ... {}, "getThumb")
 6. /var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
    OC\AppFramework\App::main("OCA\\FaceRecogn ... r", "getThumb", OC\AppFramework\ ... {}, {id: "2621",size ... "})
 7. <<closure>>
    OC\AppFramework\Routing\RouteActionHandler->__invoke({id: "2621",size ... "})
 8. /var/www/html/lib/private/Route/Router.php line 297
    call_user_func(OC\AppFramework\ ... {}, {id: "2621",size ... "})
 9. /var/www/html/lib/base.php line 1011
    OC\Route\Router->match("/apps/facerecog ... 8")
10. /var/www/html/index.php line 37
    OC::handleRequest()

GET /apps/facerecognition/face/2621/thumb/128
```
</details>

**Comment by matiasdelellis** [comment]
Good catch.. Tomorow see that.

Thanks..

**Comment by matiasdelellis** [comment]
Hi @Derkades 
Although the report seems logical to me, I cannot reproduce it. 
There is something else that is escaping me .. :thinking:

**Comment by matiasdelellis** [comment]
Hi @Derkades 
I was trying to fix another problem, and I finally understood your report .... :sweat_smile: 

Well, this must be fixed by #363 
Please. See that.
